fs/layer/node_test.go

```diff
@@ -307,8 +317,25 @@ func TestExistence(t *testing.T) {
 	}
 }
 
-func getRootNode(t *testing.T, r *estargz.Reader) *node {
-	rootNode, err := newNode(testStateLayerDigest, &testReader{r}, &testBlobState{10, 5})
+func newReader(sr *io.SectionReader) (_ *metadata.Reader, closeFn func(), _ error) {
+	f, err := ioutil.TempFile("", "layertestdb")
+	if err != nil {
+		return nil, nil, err
+	}
+	db, err := bolt.Open(f.Name(), 0666, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	closeFn = func() { db.Close(); os.Remove(f.Name()); f.Close() }
```

Comment:
You could call f.Close() right before bolt.Open() since bolt.Open anyway opens a new file.